src/Renderers/TitleNodeRenderer.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Docs Builder package.
+ * (c) Ryan Weaver <ryan@symfonycasts.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SymfonyDocsBuilder\Renderers;
+
+use Doctrine\RST\Environment;
+use Doctrine\RST\Nodes\TitleNode;
+use Doctrine\RST\Renderers\NodeRenderer;
+use Doctrine\RST\Templates\TemplateRenderer;
+
+class TitleNodeRenderer implements NodeRenderer
+{
+    /** @var TitleNode */
+    private $titleNode;
+
+    /** @var TemplateRenderer */
+    private $templateRenderer;
+
+    private static $idRenderedByFilename = [];
+
+    public function __construct(TitleNode $titleNode, TemplateRenderer $templateRenderer)
+    {
+        $this->titleNode = $titleNode;
+        $this->templateRenderer = $templateRenderer;
+    }
+
+    public function render(): string
+    {
+        $filename = $this->titleNode->getEnvironment()->getCurrentFileName();
+        $id = $this->titleNode->getId();
+
+        $actualCountRendered = self::$idRenderedByFilename[$filename][$id] ?? null;
```

Comment:
I find this variable name a bit confusing. If I'm not the only one, here are some alternative suggestions: `$numTitleOccurrences`, `$idUsagesCount`, `$timesSameIdWasUsed`, etc.

Also, instead of `null` we could initialize it to `0`, to the variable is always of type `int`. Thanks

Reply:
I agree, I updated to `idUsagesCount`